hnn_core/gui/_viz_manager.py

```diff
@@ -0,0 +1,575 @@
+            style={'button_color': self.viz_layout['theme_color']},
+            layout=self.viz_layout['btn'])
+        self.make_fig_button.on_click(self.add_figure)
+
+        # data
+        self.fig_idx = {"idx": 1}
+        self.figs = {}
+        self.gui_data = gui_data
+
+    @property
+    def widgets(self):
+        return {
+            "figs_output": self.figs_output,
+            "axes_config_tabs": self.axes_config_tabs,
+            "figs_tabs": self.figs_tabs,
+            "templates_dropdown": self.templates_dropdown
+        }
+
+    @property
+    def data(self):
+        return {
+            "use_ipympl": self.use_ipympl,
+            "simulations": self.gui_data["simulation_data"],
+            "fig_idx": self.fig_idx,
+            "visualization_output": self.viz_layout['visualization_output'],
+            "figs": self.figs
+        }
+
+    def reset_fig_config_tabs(self):
+        simulation_names = tuple(self.data['simulations'].keys())
+        for tab in self.axes_config_tabs.children:
+            controls = tab.children[1]
+            for ax_control in controls.children:
+                simulation_selection = ax_control.children[0]
+                simulation_selection.options = simulation_names
+
+    def compose(self):
+        with self.axes_config_output:
+            display(self.axes_config_tabs)
+        with self.figs_output:
+            display(Label(_fig_placeholder))
+
+        fig_output_container = VBox(
+            [self.figs_output], layout=self.viz_layout['visualization_window'])
+
+        config_panel = VBox([
+            Box(
+                [
+                    self.templates_dropdown,
+                    self.make_fig_button,
+                ],
+                layout=Layout(
+                    display='flex',
+                    flex_flow='column',
+                    align_items='stretch',
+                ),
+            ),
+            Label("Figure config:"),
+            self.axes_config_output,
+        ])
+        return config_panel, fig_output_container
+
+    def add_figure(self, b=None):
+        _add_figure(b, self.widgets, self.data, scale=0.97)
+
+    def _simulate_add_fig(self):
+        self.make_fig_button.click()
+
+    def _simulate_switch_fig_template(self, template_name):
+        assert template_name in fig_templates.keys(), "No such template"
+        self.templates_dropdown.value = template_name
+
+    def _simulate_delete_figure(self, fig_name):
+        tab = self.axes_config_tabs
+        titles = [tab.get_title(idx) for idx in range(len(tab.children))]
+        assert fig_name in titles
+        tab_idx = titles.index(fig_name)
+
+        self.axes_config_tabs.selected_index = tab_idx
+        close_button = self.axes_config_tabs.children[tab_idx].children[0]
+        close_button.click()
+
+    def _simulate_edit_figure(self, fig_name, ax_name, simulation_name,
+                              plot_type, config, operation):
```

Comment:
`config` is not a great variable name ... it's usually a catch-all for a data structure with unknown keys. Can you document this?